deepspeed/runtime/engine.py

```diff
@@ -2641,7 +2647,7 @@ def load_checkpoint(self,
                     )
                     return None, None
 
-        if self.zero_optimization_partition_weights():
+        if self.optimizer and hasattr(self.optimizer, 'checkpoint_event_prologue'):
```

Comment:
I see this repeated several times in the added code. Perhaps create a function like `self.optimizer_has_ckpt_event_prologue()`?

Reply:
Good point